### DISCUSSION TITLE
Homography values in `homography` dataset

### DISCUSSION THREAD

**Discussion by karlosos** [DonsetPG/narya#22] [Q&A] [answered]
I'm trying to use homographies that are defined in files like `homography/test_homo/0_homo.py` but have some problems with them. What transformation such homography describes? For sure it's not between source image and template.

![image](https://user-images.githubusercontent.com/3882385/109954563-c6d0f180-7ce1-11eb-900b-b2123ff806b2.png)

After some digging, I found that you're using those homographies to find 4 edge points of a football pitch. Code is here: 
https://github.com/DonsetPG/narya/blob/61096e0bc76470ddac65192ad07dfd55e059a010/narya/utils/homography.py#L320:L350

What is the source of those points?

```
np.array([[-0.5, 0.1], [-0.5, 0.5], [0.5, 0.5], [0.5, 0.1]], dtype=np.float32)
```

**Comment by karlosos** [comment, marked as answer]
Oh ok. Everything works fine. There's a need to convert from np image to torch image like in [Colab notebook](https://colab.research.google.com/drive/1VC3yd_M4va86N0q9NsYT0ajhCZ-k1sBO?usp=sharing).

```
    file_name = "0"
    image = load_image(f"../data/homography/test_img/{file_name}.jpg")
    homo = np.load(f"../data/homography/test_homo/{file_name}_homo.npy")
    warp = warp_image(np_img_to_torch_img(template), to_torch(homo), method='torch')
    warp = torch_img_to_np_img(warp[0])

    visualize(
        warp=warp,
        template=template,
        image=image,
    )
```
![image](https://user-images.githubusercontent.com/3882385/109957784-b28ef380-7ce5-11eb-97b7-746b5e61167e.png)

**Comment by DonsetPG** [comment]
Hey,
Yes that's the right way to go. 
The torch translation comes from the following paper and code:

- https://arxiv.org/pdf/1909.08034.pdf

- https://github.com/vcg-uvic/sportsfield_release

I always found them a bit odd but it works fine.

**Reply by karlosos** [reply]
Thanks! I have a follow-up question. How do you compute IOU in terms of homography? You describe it as:

> The homography estimation models use the IoU between the warped field with our estimation and ground truth homography as the primary metric.
> <cite>[Evaluating Soccer Player: from Live Camera to Deep Reinforcement Learning][1]</cite>

[1]: https://arxiv.org/abs/2101.05388

Am I understanding this correctly? You calculate IoU between those images:

![image](https://user-images.githubusercontent.com/3882385/109960623-4f06c500-7ce9-11eb-9680-367ca573023f.png)

Or maybe you have a different approach and you calculate IoU between images like that:

![image](https://user-images.githubusercontent.com/3882385/109960788-7e1d3680-7ce9-11eb-99d7-ccea644ee4dc.png)
